src/cbmc/cbmc_parse_options.cpp

```diff
@@ -922,6 +923,8 @@ void cbmc_parse_optionst::help()
     " --round-to-plus-inf          rounding towards plus infinity\n"
     " --round-to-minus-inf         rounding towards minus infinity\n"
     " --round-to-zero              rounding towards zero\n"
+    " --max-nondet-tree-depth N    limit size of nondet (e.g. input) object tree;\n" /* NOLINT(*) */ \
```

Comment:
We should have some more detailed description of how this behaves in our docs

Reply:
I've updated the documentation in `object_factory_parameters.h` for the `max_nondet_tree_depth` parameter.

Reply:
Definitely agree that this could do with some user documentation and not just source code documentation.